### PR TITLE
Cerchie/follow up schema errs class

### PR DESCRIPTION
**Pull request by Cerchie**
## Summary of Changes

@rohitsanj  and @flippingbits, this PR includes suggested changes as a follow-on to https://github.com/confluentinc/ide-sidecar/pull/178

## Any additional details or context that should be provided?

It also includes changes to DirectConnectionState to reflect an apparent refactor in the `/models` folder. 

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

- Tests:
    - [ ] Added new
    - [x] Updated existing
    - [ ] Deleted existing
- [x] Have you validated this change locally against a running instance of the Quarkus dev server?
    ```shell
    make quarkus-dev
    ```
- [x] Have you validated this change against a locally running native executable?
    ```shell
    make mvn-package-native && ./target/ide-sidecar-*-runner
    ```

### CODE REVIEW

File: src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java [hunk 1]
```diff
@@ -9,39 +9,58 @@
...
+ * It uses Caffeine cache to store and retrieve schema errors based on connection and schema IDs.
+ *
+ * <p>This class is application-scoped and observes connection lifecycle events to manage the cache.
+ * It provides methods to read, write, and clear schema errors for specific connections.
+ */
 
 @ApplicationScoped
 public class SchemaErrors {
 
+  // `ConnectionId` represents a unique connection identifier.
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java:25]: These and most of the other line comments added in this file are probably better captured as JavaDoc comments.

File: src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java [hunk 2]
```diff
@@ -9,39 +9,58 @@
...
   public record SchemaId(String clusterId, int schemaId) {
 
   }
 
+  //`Error` represents an error message associated with a schema.
   public record Error(String message) {
 
   }
+  @ConfigProperty(name = "ide-sidecar.api.host")
+  String sidecarHost;
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java:40]: How is this used?

File: src/main/java/io/confluent/idesidecar/restapi/messageviewer/MessageViewerClusterInfoProcessor.java [hunk 3]
```diff
@@ -79,25 +79,4 @@ public Future<MessageViewerContext> process(MessageViewerContext context) {
...
-
-  /**
-   * Respond to the connection being disconnected by clearing cached information.
-   *
-   * @param connection the connection that was disconnected
-   */
-  void onConnectionDisconnected(
-      @ObservesAsync @Lifecycle.Disconnected ConnectionState connection
-  ) {
-    schemaErrors.clearByConnectionId(new SchemaErrors.ConnectionId(connection.getId()));
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/messageviewer/MessageViewerClusterInfoProcessor.java:91]: The `schemaErrors` field is no longer used and can be removed.

File: src/main/java/io/confluent/idesidecar/restapi/proxy/ProxyContext.java [hunk 4]
```diff
@@ -25,7 +25,7 @@ public class ProxyContext {
   final Buffer requestBody;
   final Map<String, String> requestPathParams;
 
-  final @Nullable String connectionId;
+  final String connectionId;
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/proxy/ProxyContext.java:28]: Why this change? Are we changing the contract, or is this a correction (e.g., the field is not really allowed to be null)?

File: src/test/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializerTest.java [hunk 5]
```diff
@@ -363,7 +363,7 @@ record TestCase(
     ) {
       static TestCase nonRetryable(int statusCode) {
         // Expect only 1 try when the status code is non-retryable
-        return new TestCase(statusCode, 1, 1, null);
+        return new TestCase(statusCode, 3, 1, null);
```
Comment (rhauch) [inline src/test/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializerTest.java:366]: I'm curious -- the second parameter is `maxRetries`. Why are we changing this from 1 to 3, when (per the comment) we "expect only 1 try"?

File: src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java [hunk 6]
```diff
@@ -9,39 +9,58 @@
 import java.time.Duration;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+/**
+ * The `SchemaErrors` class manages schema error caching for different connections.
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java:15]: Backticks don't have any meaning in JavaDoc. Normal convention is to just type the class name as normal text content (suggested), or a link such as `{@link SchemaErrors}` if you *really* want it to be aligned with the type (not really recommended here since it is the JavaDoc for that type).
```suggestion
 * The SchemaErrors class manages schema error caching for different connections.
```

**Review by rhauch — COMMENTED** [review]
Thanks, @Cerchie. Overall looks good, but I do have a number of mostly minor suggestions.

File: src/main/java/io/confluent/idesidecar/restapi/connections/DirectConnectionState.java [hunk 7]
```diff
@@ -199,8 +199,8 @@ protected Future<KafkaClusterStatus> getKafkaConnectionStatus() {
         }
     ).orElseGet(
         () -> {
-            // There is no Kafka cluster configuration, so return a null Kafka status
-            return Future.succeededFuture(null);
+          // There is no Kafka cluster configuration, so return a null Kafka status
+          return Future.succeededFuture(null);
```
Comment (rohitsanj) [inline src/main/java/io/confluent/idesidecar/restapi/connections/DirectConnectionState.java:203]: This diff is extraneous and may be reverted.

File: src/main/java/io/confluent/idesidecar/restapi/connections/DirectConnectionState.java [hunk 8]
```diff
@@ -356,4 +356,4 @@ public static Throwable unwrap(Throwable t) {
     }
     return t;
   }
-}
+}
```
Comment (rohitsanj) [inline src/main/java/io/confluent/idesidecar/restapi/connections/DirectConnectionState.java:359]: This diff is extraneous and may be reverted. (A new line must be added back to the end of the file)

File: src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java [hunk 9]
```diff
@@ -9,39 +9,58 @@
...
+ * It uses Caffeine cache to store and retrieve schema errors based on connection and schema IDs.
+ *
+ * <p>This class is application-scoped and observes connection lifecycle events to manage the cache.
+ * It provides methods to read, write, and clear schema errors for specific connections.
+ */
```
Comment (rohitsanj) [inline src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java:21]: nit: The Javadoc and the class definition should not be separated by whitespace.

**Review by rohitsanj — CHANGES_REQUESTED** [review]
Hi @Cerchie, apart from my offline comment regarding the `DirectConnectionState` refactor which you've already addressed, I have a couple of very minor suggestions. Thanks!

File: src/main/java/io/confluent/idesidecar/restapi/proxy/ProxyContext.java (hunk 4 above)
Comment (Cerchie) [reply src/main/java/io/confluent/idesidecar/restapi/proxy/ProxyContext.java:28]: This was something I'd added in, probably to make a test pass, but no longer needed, so I reverted it to the prior state

File: src/test/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializerTest.java (hunk 5 above)
Comment (Cerchie) [reply src/test/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializerTest.java:366]: Again, this was a change that slipped in, probably while working on testing: https://github.com/confluentinc/ide-sidecar/pull/178/files#r1861315837

Comment (Cerchie) [reply src/test/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializerTest.java:366]: PS-- the second param is `expectedTries`.

**Comment by Cerchie** [comment]
thank you @rohitsanj  and @rhauch for the review! pushed some cleanup to address the comments

File: src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java [hunk 10]
```diff
@@ -9,6 +9,35 @@
...
+ * </ul>
+ *
+ * <p>Methods:
+ * <ul>
+ *   <li>`getSubCache(ConnectionId key)`: Retrieves or creates a sub-cache for a specific connection.</li>
+ *   <li>`readSchemaIdByConnectionId(ConnectionId connectionId, SchemaId schemaId)`: Reads a schema error from the cache.</li>
+ *   <li>`writeSchemaIdByConnectionId(ConnectionId connectionId, SchemaId schemaId, Error error)`: Writes a schema error to the cache.</li>
+ *   <li>`clearByConnectionId(ConnectionId connectionId)`: Clears the cache for a specific connection.</li>
+ *   <li>`onConnectionChange(ConnectionState connection)`: Removes the cache by connections id in response to lifecycle events..</li>
+ * </ul>
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java:39]: nit: It's more idiomatic to just put these as short JavaDoc comments on each nested type and (public) method, rather than here in the JavaDoc for the class.

File: src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java [hunk 11]
```diff
@@ -9,6 +9,35 @@
 import java.time.Duration;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+/**
+ * The SchemaErrors class manages schema error caching for different connections.
+ * It uses Caffeine cache to store and retrieve schema errors based on connection and schema IDs.
+ * <p>This class is application-scoped and observes connection lifecycle events to manage the cache.
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java:17]: The linter should complain about `<p>` not being preceded by a blank line, so this is correct:
```suggestion
 *
 * <p>This class is application-scoped and observes connection lifecycle events to manage the cache.
```

File: src/test/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializerTest.java (hunk 5 above)
Comment (rhauch) [reply src/test/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializerTest.java:366]: Hmm, I think the second parameter is `maxRetries` per [these lines](https://github.com/confluentinc/ide-sidecar/blob/7574242da2435a9aab6e68e1ac91a5be41473f42/src/test/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializerTest.java#L358-L363) just above this change:
```
    record TestCase(
        int statusCode,
        int maxRetries,
        int expectedTries,
        Boolean isKey
    ) {
      ...
```

**Review by rhauch — COMMENTED** [review]
Thanks for the quick updates, @Cerchie. A few more suggestions below.

File: src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java [hunk 12]
```diff
@@ -25,23 +46,27 @@ public record Error(String message) {
...
-  public static Cache<SchemaId, Error> getSubCache(ConnectionId key) {
+ // getSubCache(ConnectionId key): Retrieves or creates a sub-cache for a specific connection.
+  public Cache<SchemaId, Error> getSubCache(ConnectionId key) {
     return cacheOfCaches.computeIfAbsent(
         key,
-        k -> Caffeine.newBuilder().expireAfterAccess(SCHEMA_FETCH_ERROR_TTL).build());
+        k -> Caffeine.newBuilder().expireAfterAccess(Duration.ofSeconds(schemaFetchErrorTtl)).build());
   }
 
+  // readSchemaIdByConnectionId(ConnectionId connectionId, SchemaId schemaId): Reads a schema error from the cache.
```
Comment (rohitsanj) [inline src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java:61]: We typically try to use [Javadocs](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html) style comments to document functionality. The comment used here `//` is a [single-line (or end-of-line) comment](https://www.oracle.com/java/technologies/javase/codeconventions-comments.html) and we reserve those for clarifying explanations embedded in code implementations.

```
/**
* Reads a schema error from the cache.
*/
```

Comment (Cerchie) [reply src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java]: thanks! pushed a change

File: src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java [hunk 13]
```diff
@@ -9,6 +9,27 @@
...
+ * <ul>
+ *   <li>`ide-sidecar.schema-fetch-error-ttl`: The time-to-live duration for schema fetch errors in the cache.</li>
+ * </ul>
+ *
+ * <p>Records:
+ * <ul>
+ *   <li>`ConnectionId`: Represents a unique connection identifier.</li>
+ *   <li>`SchemaId`: Represents a unique schema identifier within a cluster.</li>
+ *   <li>`Error`: Represents an error message associated with a schema.</li>
+ * </ul>
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java:31]: Can we please move these to JavaDoc on the record types themselves?

File: src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java [hunk 14]
```diff
@@ -25,23 +46,44 @@ public record Error(String message) {
 
   }
 
-  private static final Duration SCHEMA_FETCH_ERROR_TTL = Duration.ofSeconds(30);
+  @ConfigProperty(name = "ide-sidecar.schema-fetch-error-ttl")
+  long schemaFetchErrorTtl;
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java:50]: This still can be static and final, and we can still have this be a `Duration` type. See other examples in the codebase.

File: src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java [hunk 15]
```diff
@@ -61,7 +61,7 @@ public class RecordDeserializer {
       .asMap("ide-sidecar.serde-configs");
   private final int schemaFetchMaxRetries;
 
-  SchemaErrors schemaErrors;
+  private final SchemaErrors schemaErrors;
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java:64]: Nit: static fields should be separated from and before non-final fields.

File: src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java [hunk 16]
```diff
@@ -25,23 +46,44 @@ public record Error(String message) {
...
+   * @param connectionId The connection identifier.
+   * @param schemaId     The schema identifier.
+   * @return The schema error, or null if not found.
+   */
   public Error readSchemaIdByConnectionId(
       ConnectionId connectionId,
       SchemaId schemaId
   ) {
     return getSubCache(connectionId).getIfPresent(schemaId);
   }
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java:78]: The record types primarily exist to simplify the definition of the map of maps, so we can be don't need to expose them outside the `SchemaErrors` class. Having these methods _require_ the record types makes these methods harder to use. So we could either:
1. Refactor the public methods to not use the record types; or
2. Add additional methods that don't use record types

For example, option 2 would add methods like:
```
  public Error readSchemaIdByConnectionId(String connectionId, String clusterId, String schemaId) {
    return readSchemaIdByConnectionId(
        new ConnectionId(connectionId),
        new SchemaId(clusterId, schemaId)
    );
  }
```

File: src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java [hunk 17]
```diff
@@ -225,15 +225,14 @@ public DecodedResult deserialize(
...
     if (schemaErrors.readSchemaIdByConnectionId(new SchemaErrors.ConnectionId(context.getConnectionId()), schemaId) != null) {
       return new DecodedResult(
           // If the schema fetch failed, we can't decode the data, so we just return the raw bytes.
           // We apply the encoderOnFailure function to the bytes before returning them.
           onFailure(encoderOnFailure, bytes),
-          String.valueOf(
-              schemaErrors.readSchemaIdByConnectionId(new SchemaErrors.ConnectionId(context.getConnectionId()), schemaId))
-      );
+          schemaErrors.readSchemaIdByConnectionId(connectionId, schemaId).message());
     }
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java:236]: We really should look this up once and use the result multiple times. If we read multiple times, there's a chance that we'll get different results back.
```
var schemaId = getSchemaIdFromRawBytes(bytes);
var error = schemaErrors.readSchemaIdByConnectionId(
    contex.getConnectionId(),
    context.getClusterId(),
    schemaId
);
if (error != null) {
  return new DecodedResult(
      onFailure(encoderOnFailure, bytes),
      error.message()
  );
}
```

**Review by rhauch — COMMENTED** [review]
Thanks for the updates, @Cerchie.

**Review by rohitsanj — APPROVED** [review]
Thanks for the updates! LGTM.

Let's wait for @rhauch to bless as well.

File: src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java (hunk 16 above)
Comment (Cerchie) [reply src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java:78]: I might be missing something... wouldn't this recurse as written?

Comment (rhauch) [reply src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java:78]: No, the `readSchemaIdByConnectionId(String, String, String)` method can call the `readSchemaIdByConnectionId(ConnectionId, SchemaId)` method and it will not recurse since they're different signatures.

**Comment by Cerchie** [comment]
> Thanks for the updates, @Cerchie.

Thank you! with the exception of resolving the issue of exposing the records outside the class, I've addressed the comments and reviewed a diff to try to catch style issues before pushing up the latest commits.

File: src/main/resources/application.yml [hunk 18]
```diff
@@ -67,6 +68,7 @@ vscode:
     version: unknown
   version: unknown
 ide-sidecar:
+  schema-fetch-error-ttl: 30
```
Comment (rhauch) [inline src/main/resources/application.yml:71]: Nit: can we put this lower, maybe just before `schema-fetch-retry` (since before it would be in proper sort-order)?

File: src/main/resources/application.yml [hunk 19]
```diff
@@ -2,6 +2,7 @@
 ---
 "%test":
   ide-sidecar:
+    schema-fetch-error-ttl: 30
```
Comment (rhauch) [inline src/main/resources/application.yml:5]: Nit: can we put this lower, maybe just before `schema-fetch-retry` (since before it would be in proper sort-order)?

File: src/test/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializerTest.java [hunk 20]
```diff
@@ -46,7 +46,7 @@ public class RecordDeserializerTest {
 
   private static final SchemaErrors.SchemaId VALID_SCHEMA_ID = new SchemaErrors.SchemaId("fake_cluster_id", 10008);
 
-  SchemaErrors.ConnectionId CONNECTION_1_ID = new SchemaErrors.ConnectionId("c1");
+  String CONNECTION_1_ID = "c1";
```
Comment (rhauch) [inline src/test/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializerTest.java:49]: We're considering this a constant if the name is in all caps, so let's make it final
```suggestion
  static final String CONNECTION_1_ID = "c1";
```

File: src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java [hunk 21]
```diff
@@ -345,7 +350,12 @@ private void cacheSchemaFetchError(
         schemaId.schemaId(),
         e.getMessage()
     ));
-    schemaErrors.writeSchemaIdByConnectionId(new SchemaErrors.ConnectionId(context.getConnectionId()), schemaId, errorMessage);
+    schemaErrors.writeSchemaIdByConnectionId(
+        context.getConnectionId(),
+        String.valueOf(schemaId.schemaId()),
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java:355]: I think this was a miss on my part, but we shouldn't really be converting the schema IDs to strings and then back to integers. If we need them as integers, why don't we expect an integer parameter?

File: src/test/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializerTest.java [hunk 22]
```diff
@@ -63,7 +63,7 @@ public class RecordDeserializerTest {
       null,
       null,
       null,
-      CONNECTION_1_ID,
+      new SchemaErrors.ConnectionId(CONNECTION_1_ID),
```
Comment (rhauch) [inline src/test/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializerTest.java:66]: I recently suggested think NOT exposing `SchemaErrors.ConnectionId` or `SchemaErrors.SchemaId` outside of `SchemaErrors`. If that's the case, why do we need (or want) the MessageViewerContext to have an instance of ``SchemaErrors.ConnectionId`?

I think it breaks encapsulation and abstraction if `MessageViewerContext` has a field of type `SchemaErrors.ConnectionId` just to hold onto a connection ID, as the connection ID is not at all related to SchemaErrors.

File: src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java [hunk 23]
```diff
@@ -225,14 +225,19 @@ public DecodedResult deserialize(
       return result.get();
     }
     SchemaErrors.SchemaId schemaId = new SchemaErrors.SchemaId(context.getClusterId(), getSchemaIdFromRawBytes(bytes));
+    SchemaErrors.ConnectionId connectionId = new SchemaErrors.ConnectionId(context.getConnectionId());
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java:228]: Again, why does `RecordDeserializer` know anything about `SchemaErrors.SchemaId` or `SchemaErrors.ConnectionId`? Those types should be _internal_ / _private_ to the `SchemaErrors` class.

File: src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java [hunk 24]
```diff
@@ -9,54 +9,124 @@
...
+  public static final Map<ConnectionId, Cache<SchemaId, Error>> cacheOfCaches = new ConcurrentHashMap<>();
 
-  public static Cache<SchemaId, Error> getSubCache(ConnectionId key) {
+  /**
+   * Retrieves or creates a sub-cache for a specific connection.
+   *
+   * @param key The connection identifier.
+   * @return The sub-cache for the connection.
+   */
+  public Cache<SchemaId, Error> getSubCache(ConnectionId key) {
```
Comment (rhauch) [inline src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java:67]: Why is this method public? IIUC, it's not used outside of this class, so we can do this:
```suggestion
  private Cache<SchemaId, Error> getSubCache(ConnectionId key) {
```
If it were used in testing, we could make it _package protected_.

**Review by rhauch — COMMENTED** [review]
A few more suggestions. The big question is: why are the `ConnectionId` and `SchemaId` record types even exposed outside of `SchemaErrors`. As mentioned before, these types only exist to make the implementation of `SchemaErrors` more clear. 
We only added them because it's very confusing to have `Map<String, Map<String, String>` with the second-level key be a string composed from two other strings. Using record types makes this much clearer: `Map<ConnectionId, Map<SchemaId, Error>`.

And a recent suggestion was to remove the `ConnectionId` and `SchemaId` from the public method signatures, _so that we can make these record types private_. We haven't yet made the record types private, and need to do that.

The `Error` record type can be public, because it valuable outside to the components that use the `SchemaErrors` object.

One might argue that we should have used a record type for connection ID, but if we were to do that we'd want that record type to be defined in a common place and definitely NOT scoped within `SchemaErrors`, because we'd want to use it *everywhere*. But that's a big change, and at this point it's not worth it as we don't need the type safety everywhere in the codebase. (As mentioned above, it's _locally_ valuable in `SchemaErrors` to help disambiguate the types in the nested caches.)

**Comment by Cerchie** [comment]
> A few more suggestions. The big question is: why are the `ConnectionId` and `SchemaId` record types even exposed outside of `SchemaErrors`. As mentioned before, these types only exist to make the implementation of `SchemaErrors` more clear. We only added them because it's very confusing to have `Map<String, Map<String, String>` with the second-level key be a string composed from two other strings. Using record types makes this much clearer: `Map<ConnectionId, Map<SchemaId, Error>`.
> 
> And a recent suggestion was to remove the `ConnectionId` and `SchemaId` from the public method signatures, _so that we can make these record types private_. We haven't yet made the record types private, and need to do that.
> 
> The `Error` record type can be public, because it valuable outside to the components that use the `SchemaErrors` object.
> 
> One might argue that we should have used a record type for connection ID, but if we were to do that we'd want that record type to be defined in a common place and definitely NOT scoped within `SchemaErrors`, because we'd want to use it _everywhere_. But that's a big change, and at this point it's not worth it as we don't need the type safety everywhere in the codebase. (As mentioned above, it's _locally_ valuable in `SchemaErrors` to help disambiguate the types in the nested caches.)

Ah I missed making those records private after adjusting the methods. Thanks for your patience. Moved the ttl setting to a more appropriate place, made the records private, and also made the `schemaId` an int.

File: src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java (hunk 9 above)
Comment (rhauch) [reply src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java:26]: +1

File: src/test/java/io/confluent/idesidecar/restapi/integration/AbstractIT.java [hunk 25]
```diff
@@ -63,15 +63,15 @@
  */
 public abstract class AbstractIT extends SidecarClient implements ITSuite {
 
-  static SchemaErrors.ConnectionId VALID_CONNECTION_ID = new SchemaErrors.ConnectionId("c1");
+  private static String VALID_CONNECTION_ID = "c1";
```
Comment (rhauch) [inline src/test/java/io/confluent/idesidecar/restapi/integration/AbstractIT.java:66]:
```suggestion
  private static final String VALID_CONNECTION_ID = "c1";
```

**Review by rhauch — COMMENTED** [review]
Two super minor nits, but otherwise looks great.

**Review by rhauch — APPROVED** [review]
LGTM. Nice work, @Cerchie!
